### PR TITLE
Local sshkey field added to local-users.j2 to enable sshkey definitio…

### PR DESCRIPTION
**Pull request by sengarth**
## Change Summary

Add "sshkey" field in local-users.j2 to allow specification of ssh key for user accounts. This allow configuration of the local sshkey via eos_cli_config_gen. Prior to this, custom templates or custom eos_cli is needed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Component(s) name
`arista.avd.eos_cli_config_gen`

## Proposed changes

When running the user creation template "local-users.j2":
1. Test whether "local_users[local_user].sshkey is arista.avd.defined"
2. If yes, create CLI "username xxx ssh-key local_users[local_user].sshkey" 
3. Generate output to config file

```
local_users:
  < user_1 >:
    privilege: < 1-15 >
    role: < role >
    sha512_password: "< sha_512_password >"
    no_password: < true | do not configure a password for given username. sha512_password MUST not be defined for this user. >
    ssh_key: "< ssh-key_string >"

  < user_2 >:
    privilege: < 1-15 >
    role: < role >
    sha512_password: "< sha_512_password >"
    no_password: < true | do not configure a password for given username. sha512_password MUST not be defined for this user. >
    ssh_key: "< ssh-key_string >"
```
## How to test

molecule test --scenario-name eos_cli_config_gen

Using prescribed environment from ansible-avd via:
make install-requirements
make install-docker (but for Ubuntu focal)

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -345,11 +345,13 @@ local_users:
     role: < role >
     sha512_password: "< sha_512_password >"
     no_password: < true | do not configure a password for given username. sha512_password MUST not be defined for this user. >
+    sshkey: "< ssh-key_string >"
   < user_2 >:
     privilege: < 1-15 >
     role: < role >
     sha512_password: "< sha_512_password >"
     no_password: < true | do not configure a password for given username. sha512_password MUST not be defined for this user. >
+    sshkey: "< ssh-key_string >"
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: to be consistent with the rest of the data model, could you update `sshkey` to `ssh_key`. 
We have tried to apply conversion from `-` in EOS to `_` in YAML as much as possible.

**Review by titom73 — CHANGES_REQUESTED** [review]
Thanks for this nice first PR!
Once the variable name will be updated, we will be more than happy to merge it as soon as possible !

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (sengarth) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Both doc and YAML have been updated to reflect the appropriate naming convention. Thanks for the heads up, will take note of this in future.

Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Great !

**Review by titom73 — APPROVED** [review]

**Pull request merged**